src/GraphQL/TypeApi.hs

```diff
@@ -87,14 +93,18 @@ instance forall m hg. (MonadThrow m, MonadIO m, HasGraph m hg) => HasGraph m (Li
     let a = traverse (flip (buildResolver @m @hg) selectionSet) handler
     in map GValue.toValue a
 
+instance forall m ks enum. (MonadThrow m, MonadIO m, GraphQLEnum enum) => HasGraph m (Enum ks enum) where
+  type HandlerType m (Enum ks enum) = enum
+  buildResolver handler _ = pure (enumToValue handler)
+
+
 -- TODO: lookup is O(N^2) in number of arguments (we linearly search
 -- each argument in the list) but considering the graphql use case
 -- where N usually < 10 this is probably OK.
--- TODO (Maybe Int) types that can convert Nothing
-lookupValue :: AST.Name -> [AST.Argument] -> Either Text AST.Value
+lookupValue :: AST.Name -> [AST.Argument] -> Maybe AST.Value
 lookupValue name args = case find (\(AST.Argument name' _) -> name' == name) args of
-  Nothing -> Left ("Argument not found:" <> name)
-  Just (AST.Argument _ value) -> pure value
+  Nothing -> Nothing
+  Just (AST.Argument _ value) -> Just value
```

Comment:
classic lens territory but I didn't feel like pulling in lens for this API. Without lens I find the local pattern match more readable (no indirection through really small helpers)